File: radian size/radian.js
const canvas = document.getElementById('circleCanvas');
const ctx = canvas.getContext('2d');
const radius = 150;

function drawCircle() {
    ctx.clearRect(0, 0, canvas.width, canvas.height);
    ctx.beginPath();
    ctx.arc(canvas.width / 2, canvas.height / 2, radius, 0, 2 * Math.PI);
    ctx.stroke();
}

function drawAngle() {
    const degrees = document.getElementById('angleInput').value;
    if (degrees < 0 || degrees > 360 || isNaN(degrees)) {
        alert('Please enter a valid angle between 0 and 360 degrees.');
        return;
    }
    const radians = degrees * (Math.PI / 180);

    drawCircle();
    ctx.beginPath();
    ctx.moveTo(canvas.width / 2, canvas.height / 2);
    ctx.lineTo(canvas.width / 2 + radius * Math.cos(0), canvas.height / 2 + radius * Math.sin(0));
    ctx.stroke();

    ctx.beginPath();
    ctx.moveTo(canvas.width / 2, canvas.height / 2);
    ctx.lineTo(canvas.width / 2 + radius * Math.cos(radians), canvas.height / 2 + radius * Math.sin(radians));
    ctx.stroke();
    
    ctx.font = '16px Arial';
    ctx.fillText(`${degrees} degrees = ${radians.toFixed(2)} radians`, 10, canvas.height - 10);
}

drawCircle();

File: radian size/radian.py
import turtle
import math

# Setup screen
screen = turtle.Screen()
screen.title("Understanding Radians")
screen.bgcolor("lightblue")

# Setup turtle
t = turtle.Turtle()
t.speed(1)

# Function to draw a circle and the radius
def draw_circle(radius):
    t.penup()
    t.goto(0, -radius)
    t.pendown()
    t.circle(radius)
    t.penup()
    t.goto(0, 0)
    t.pendown()

# Function to draw an angle
def draw_angle(degrees):
    radians = math.radians(degrees)
    arc_length = radius * radians
    t.penup()
    t.goto(0, 0)
    t.setheading(0)
    t.pendown()
    t.forward(radius)
    t.left(90)
    t.circle(radius, degrees)
    t.penup()
    t.goto(0, 0)
    t.setheading(0)
    t.pendown()
    t.forward(radius)
    t.write(f"{degrees} degrees = {round(radians, 2)} radians", align="left", font=("Arial", 12, "normal"))

# Main game loop
def main():
    global radius
    radius = 100
    draw_circle(radius)

    while True:
        try:
            degrees = int(screen.textinput("Input", "Enter an angle in degrees (0 to 360):"))
            if 0 <= degrees <= 360:
                t.clear()
                draw_circle(radius)
                draw_angle(degrees)
            else:
                t.write("Please enter a value between 0 and 360", align="center", font=("Arial", 12, "normal"))
        except:
            break

# Start the game
main()
screen.mainloop()